pitop/labs/webserver/static/joysticks.js

```diff
@@ -0,0 +1,106 @@
+var cmd_vel_twist = {
+  'linear': {
+    'x': 0.0
+  },
+  'angular': {
+    'z': 0.0
+  }
+};
+
+var pan_tilt_twist = {
+  'angular': {
+    'y': 0.0,
+    'z': 0.0
+  }
+};
+
+const MAX_LINEAR_SPEED = 0.44;
+const MAX_ANGULAR_SPEED = 5.12;
+const MAX_SERVO_SPEED = 100;
+var publishCmdVelImmediately = true;
+var publishPanTiltImmediately = true;
+
+function cmdVelTwistPublisher(linear, angular) {
+    if (linear !== undefined && angular !== undefined) {
+        cmd_vel_twist.linear.x = linear;
+        cmd_vel_twist.angular.z = angular;
+    } else {
+        cmd_vel_twist.linear.x = 0;
+        cmd_vel_twist.angular.z = 0;
+    }
+    // console.log(linear)
+    window.command['cmd_vel'](cmd_vel_twist);
+}
+
+function panTiltTwistPublisher(angular_y, angular_z) {
+    if (angular_y !== undefined && angular_z !== undefined) {
+        pan_tilt_twist.angular.y = angular_y;
+        pan_tilt_twist.angular.z = angular_z;
+    } else {
+        pan_tilt_twist.linear.x = 0;
+        pan_tilt_twist.angular.z = 0;
+    }
+    // console.log(linear)
+    window.command['pan_tilt'](pan_tilt_twist);
+}
+
+['left', 'right'].forEach((position) => {
+    const joystickZone = document.getElementById(position + 'JoystickContainer');
+
+    const joystick = nipplejs.create({
+        zone: joystickZone,
+        mode: 'static',
+        size: 200,
+        position: {top: 50, [position]: 50}
+    }).get();
+
+    joystick.on('move', (evt, joystick) => {
+
+        let direction = joystick.angle.degree - 90;
+        if (direction > 180) {
+            direction = -(450 - joystick.angle.degree);
+        }
+
+        if (position === "left") {
+            // joystick distance max is 100, so set that to be maximum speeds
+            let linear = Math.cos(direction / 57.29) * joystick.distance * MAX_LINEAR_SPEED / 100.0;
+            let angular = Math.sin(direction / 57.29) * joystick.distance * MAX_ANGULAR_SPEED / 100.0;
+            if (publishCmdVelImmediately) {
+              publishCmdVelImmediately = false;
+              cmdVelTwistPublisher(linear, angular);
+              setTimeout(function () {
+                publishCmdVelImmediately = true;
+              }, 50);
+            }
+        } else {
+            let angular_y = -Math.cos(direction / 57.29) * joystick.distance * MAX_SERVO_SPEED / 100.0;
+            let angular_z = Math.sin(direction / 57.29) * joystick.distance * MAX_SERVO_SPEED / 100.0;
```

Comment:
Change to angle_y and angle_z. Remove MAX_SERVO_SPEED and change to MAX_SERVO_ANGLE instead since this implementation uses position and not angular speed